### ISSUE TITLE
Lab screen implementation

### ISSUE THREAD

**Issue by amanda-phet** [phetsims/number-play#33]
- [x] Fix play area layering
- [x] objects snap to first open spot when dropped over ten frame
- [x] create dynamic layout for the screen
- [x] prevent mixing of objects in ten frames and don't allow ones in ten frames
- [x] ten frame shrinks and returns when dropped over ten frame creator
- [x] objects return when ten frame  dropped over creator
- [x] grabbing objects in the ten frame moves the whole ten frame instead of pulling them out\
- [x] ten frame is "selectable" and an eject button appears when the ten frame is selected
- [x] hitting the eject button returns an object to the creator nodes
- [x] rejected additions to the ten frame animate to near spot outside the ten frame
- [x] constrain dragging of objects to below number card panel
- [x] fix number card origins
- [x] constrain dragging of number cards to above the bottom panels
- [x] constrain the dragging of the ten frames to below the number panel
- [x] constrain the dragging of the symbol cards to below the number panel
- [x] make any objects dropped in bottom row return home
- [x] constrain dragging of counting object and cards to not go over symbols panel
- [ ] ~~make objects on their way out to the play area not pickable~~
- [ ] ~~new objects from buttons should animate closer to origin~~
- [x] memory management - moved to https://github.com/phetsims/number-play/issues/87

**Comment by chrisklus** [comment]
I got ambitious over the weekend (before all the issues came in from Number Play dev test), and investigated how far out of reach the lab screen is. On the labScreen branch, I converted it to TypeScript, fixed the layering problem so we can have a bunch of play areas, and did some work on the ten frames. I bailed after the ten frames started getting complicated and it was clear that this screen will have lots of design questions.
